uaclient/contract_data_types.py

```diff
@@ -0,0 +1,572 @@
+        Field("aptURL", StringDataValue, False),
+        Field("aptKey", StringDataValue, False),
+        Field("suites", data_list(StringDataValue), False),
+        Field("server", StringDataValue, False),
+        Field("remoteServer", StringDataValue, False),
+        Field("caCerts", StringDataValue, False),
+        Field("snapChannel", StringDataValue, False),
+        Field("pypiURL", StringDataValue, False),
+        Field("url", StringDataValue, False),
+    ]
+
+    def __init__(
+        self,
+        additionalPackages: Optional[List[str]],
+        aptURL: Optional[str],
+        aptKey: Optional[str],
+        suites: Optional[List[str]],
+        server: Optional[str],
+        remoteServer: Optional[str],
+        caCerts: Optional[str],
+        snapChannel: Optional[str],
+        pypiURL: Optional[str],
+        url: Optional[str],
+    ):
+        self.additionalPackages = additionalPackages
+        self.aptURL = aptURL
+        self.aptKey = aptKey
+        self.suites = suites
+        self.server = server
+        self.remoteServer = remoteServer
+        self.caCerts = caCerts
+        self.snapChannel = snapChannel
+        self.pypiURL = pypiURL
+        self.url = url
+
+
+class SeriesEntitlementsProperties(DataObject):
+    fields = [
+        Field("affordances", Affordances, False),
+        Field("obligations", Obligations, False),
+        Field("directives", Directives, False),
+    ]
+
+    def __int__(
+        self,
+        affordances: Optional[Affordances],
+        obligations: Optional[Obligations],
+        directives: Optional[Directives],
+    ):
+        self.affordances = affordances
+        self.obligations = obligations
+        self.directives = directives
+
+
+class SeriesEntitlements(DataObject):
+    fields = [
+        Field("trusty", SeriesEntitlementsProperties, False),
+        Field("xenial", SeriesEntitlementsProperties, False),
+        Field("bionic", SeriesEntitlementsProperties, False),
+        Field("focal", SeriesEntitlementsProperties, False),
+        Field("jammy", SeriesEntitlementsProperties, False),
+    ]
+
+    def __init__(
+        self,
+        trusty: Optional[SeriesEntitlementsProperties],
+        xenial: Optional[SeriesEntitlementsProperties],
+        bionic: Optional[SeriesEntitlementsProperties],
+        focal: Optional[SeriesEntitlementsProperties],
+        jammy: Optional[SeriesEntitlementsProperties],
+    ):
+        self.trusty = trusty
+        self.xenial = xenial
+        self.bionic = bionic
+        self.focal = focal
+        self.jammy = jammy
+
+
+class Entitlement(DataObject):
+    fields = [
+        Field("entitled", BoolDataValue, False),
+        Field("type", StringDataValue, False),
+        Field("affordances", Affordances, False),
+        Field("obligations", Obligations, False),
+        Field("directives", Directives, False),
+        Field("series", SeriesEntitlements, False),
```

Comment:
We can drop the `series` field as it is old.

We'll need the `overrides` field though https://github.com/canonical/ua-contracts/blob/develop/docs/contracts.yaml#L1312=

Reply:
Thanks for catching `overrides`.